portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/ExportResult.java

```diff
@@ -1,23 +1,32 @@
 package org.dataportabilityproject.spi.transfer.provider;
 
+import com.google.common.base.Preconditions;
 import org.dataportabilityproject.spi.transfer.types.ContinuationData;
 import org.dataportabilityproject.types.transfer.models.DataModel;
 
-/** The result of an item export operation. */
+/**
+ * The result of an item export operation.
+ */
 public class ExportResult<T extends DataModel> {
+
   public static final ExportResult CONTINUE = new ExportResult(ResultType.CONTINUE);
   public static final ExportResult END = new ExportResult(ResultType.CONTINUE);
+  private static final String MUST_HAVE_THROWABLE = "ExportResult with ResultType = ERROR must hold a throwable";
+
   private ResultType type;
   private String message;
   private T exportedData;
   private ContinuationData continuationData;
+  private Throwable throwable;
+
   /**
    * Ctor used to return error or retry results.
    *
    * @param type the result type
    * @param message the result message, if any
    */
   public ExportResult(ResultType type, String message) {
+    Preconditions.checkArgument(!type.equals(ResultType.ERROR), MUST_HAVE_THROWABLE);
```

Comment:
since this is done a few times, I'd move this (and the string constant) to a new method that explains what we're doing / why in its name, like "verifyNonErrorResult" or "validateResultType"